components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/handler/impl/ConfigStoreBasedTemplateHandler.java

```diff
@@ -0,0 +1,225 @@
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceTypeAdd;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resources;
+import org.wso2.carbon.identity.configuration.mgt.core.search.Condition;
+import org.wso2.carbon.identity.template.mgt.TemplateMgtConstants;
+import org.wso2.carbon.identity.template.mgt.exception.TemplateManagementException;
+import org.wso2.carbon.identity.template.mgt.function.ResourceToTemplate;
+import org.wso2.carbon.identity.template.mgt.function.TemplateToResource;
+import org.wso2.carbon.identity.template.mgt.function.TemplateToResourceAdd;
+import org.wso2.carbon.identity.template.mgt.handler.TemplateHandler;
+import org.wso2.carbon.identity.template.mgt.internal.TemplateManagerDataHolder;
+import org.wso2.carbon.identity.template.mgt.model.Template;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.getTenantDomainFromCarbonContext;
+import static org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleClientException;
+import static org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException;
+
+/**
+ * Config store based template handler.
+ */
+public class ConfigStoreBasedTemplateHandler implements TemplateHandler {
+
+    private static Log log = LogFactory.getLog(ConfigStoreBasedTemplateHandler.class);
+
+    @Override
+    public String addTemplate(Template template) throws TemplateManagementException {
+
+        return addTemplateToConfigStore(template);
+    }
+
+    @Override
+    public Template getTemplateById(String templateId) throws TemplateManagementException {
+
+        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
+        try {
+            Resource resource = configManager.getTenantResourceById(templateId);
+            Template template = new ResourceToTemplate().apply(resource);
+            if (resource.getFiles().size() == 1) {
+                InputStream templateScriptInputStream = configManager
+                        .getFileById(resource.getResourceType(), resource.getResourceName(),
+                                resource.getFiles().get(0).getId());
+                template.setTemplateScript(IOUtils.toString(templateScriptInputStream));
```

Comment:
try(InputStream templateScriptInputStream = configManager
                        .getFileById(resource.getResourceType(), resource.getResourceName(),
                                resource.getFiles().get(0).getId())) {
                template.setTemplateScript(IOUtils.toString(templateScriptInputStream));}

Reply:
IOUtils.toString does not close the stream... since this is AutoClosable above will fix that